gms/gossiper.cc

```diff
@@ -421,6 +425,120 @@ future<> gossiper::handle_echo_msg() {
+            auto nodes_to_check = boost::copy_range<std::set<gms::inet_address>>(g.endpoint_state_map | boost::adaptors::map_keys);
+            for (auto& seed : g.get_seeds()) {
+                nodes_to_check.insert(seed);
+            }
+            nodes_to_check.erase(myip);
+            auto node_to_be_replaced = g.get_node_to_be_replaced();
+            if (node_to_be_replaced) {
+                nodes_to_check.erase(*node_to_be_replaced);
+            }
+            size_t nodes_down = 0;
+            std::unordered_map<gms::inet_address, cluster_ring_status_response> queries;
+            parallel_for_each(nodes_to_check, [&nodes_down, &queries] (const auto& node) {
+                return netw::get_local_messaging_service().send_cluster_ring_status(netw::msg_addr(node)).then([node, &nodes_down, &queries] (cluster_ring_status_response result) {
+                    queries[node] = std::move(result);
+                }).handle_exception([node, &nodes_down] (std::exception_ptr ep) {
+                    try {
+                        std::rethrow_exception(ep);
+                    } catch (seastar::rpc::unknown_verb_error& e) {
+                        logger.info("Node {} does not support cluster_ring_status: {}", node, e);
+                        throw e;
+                    } catch (seastar::rpc::closed_error& e) {
+                        nodes_down++;
+                    } catch (...) {
+                        logger.info("Failed to send cluster_ring_status to {}: ", node, std::current_exception());
+                        throw;
+                    }
+                });
+            }).get();
+            size_t matched = 0;
+            logger.debug("Node {}, status = {}", myip, local.status);
+            for (auto& query : queries) {
+                cluster_ring_status_response& remote = query.second;
+                logger.debug("Node {}, status = {}", query.first, remote.status);
+                if (remote == local) {
+                    matched++;
+                }
+            }
+            size_t needed = nodes_to_check.size();
+            if (nodes_down) {
+                logger.info("Wait for cluster_ring_status, matched={}, needed={}, nodes_down={}, nodes_to_check={}, gave up due to nodes down",
+                        matched, needed, nodes_down, nodes_to_check);
+                throw std::runtime_error("Gave up wait for gossip with query becasue nodes are down");
+            }
+            if (matched == needed) {
+                logger.info("Wait for cluster_ring_status, matched={}, needed={}, nodes_down={}, nodes_to_check={}, done",
+                        matched, needed, nodes_down, nodes_to_check);
+                return;
+            }
+            logger.debug("Wait for cluster_ring_status, matched={}, needed={}, nodes_down={}, nodes_to_check={}, try again",
+                    matched, needed, nodes_down, nodes_to_check);
+            sleep_abortable(std::chrono::milliseconds(200)).get();
```

Comment:
Above you have a sleep of 100ms. Heee it's 200ms. Any particular reason why this one is higher, and we can't use 100ms for both?